esphome/components/zephyr_openthread/__init__.py

```diff
@@ -0,0 +1,243 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import gpio
+from esphome import pins
+from esphome.components.zephyr import (
+    zephyr_add_prj_conf,
+    zephyr_add_overlay,
+    zephyr_add_mcuboot_overlay,
+)
+from esphome.const import CONF_ID, CONF_ENABLE_IPV6, CONF_OTA, CONF_TRANSPORT, CONF_UDP
+from esphome.core import CORE, coroutine_with_priority, _LOGGER
+import enum
+
+from .const import (
+    CONF_CHANNEL,
+    CONF_NETWORK_KEY,
+    CONF_NETWORK_NAME,
+    CONF_PANID,
+    CONF_RADIO_TX_POWER,
+    CONF_XPANID,
+    CONF_PSKC,
+    CONF_FORCE_DATASET,
+    CONF_SHELL,
+    DEFAULT_CHANNEL,
+    DEFAULT_NETWORK_KEY,
+    DEFAULT_NETWORK_NAME,
+    DEFAULT_PANID,
+    DEFAULT_RADIO_TX_POWER,
+    DEFAULT_XPANID,
+    DEFAULT_PSKC,
+    DEFAULT_FORCE_DATASET,
+    DEFAULT_SHELL,
+    zephyr_openthread_ns,
+)
+
+from . import config_validation as ot_cv
+
+CODEOWNERS = ["@felipejfc"]
+DEPENDENCIES = ["zephyr", "nrf52"]
+AUTO_LOAD = ["mdns", "network"]
+
+OpenThreadZephyr = zephyr_openthread_ns.class_("OpenThreadZephyr", cg.Component)
+OpenThreadDNSComponent = zephyr_openthread_ns.class_(
+    "OpenThreadDNSComponent", cg.Component
+)
+
+# Define new constant for factory reset pin
+# TODO: Move this to constants, leaving it here for now to avoid breaking changes
+CONF_FACTORY_RESET_PIN = "factory_reset_pin"
+
+CONF_DEVICE_TYPE = "device_type"
+# Define enum for device types
+class DeviceType(enum.Enum):
+    MTD = "mtd"
+    FTD = "ftd"
+
+DEFAULT_DEVICE_TYPE = DeviceType.FTD
```

Comment:
Getting an error:
`Unknown value 'DeviceType.FTD', valid options are 'mtd', 'ftd'`.

The error comes from using an enum value as the default for a field validated by cv.enum. Set the default to the string key ("ftd") so cv.enum can map it to DeviceType.FTD.

```suggestion
DEFAULT_DEVICE_TYPE = "ftd"
```